packages/core/src/ObjectContainer.js

```diff
@@ -484,14 +484,28 @@ export default class ObjectContainer {
    *         implementation is known to this object container.
    */
   _getEntry(name) {
+    let nameToGet = name;
+
+    if (Array.isArray(name)) {
+      nameToGet = name?.[0];
+    }
+
+    if (this._isSpread(name)) {
+      nameToGet = nameToGet.replace('...', '');
+    }
+
+    if (this._isOptional(nameToGet)) {
+      nameToGet = nameToGet.replace('?', '');
+    }
+
```

Comment:
You probably don't have to check for optional or spread in this case, but just normalize the name (e.g. strip all meta symbols).

So all the ifs above can be replaced with:

```javascript
let entryName = Array.isArray(name) ? name : name[0];

// Remove all meta symbols from the start of the alias
if (typeof entryName === 'string') {
  entryName = entryName.replace(SPREAD_RE, '');
  entryName = entryName.replace(OPTIONAL_RE, '');
}

// ... These should be fined at top of the file -->
const SPREAD_RE = /^\.../;
const OPTIONAL_RE = /^\?/;
```

Since you check anyway for spread an optional types in following methods and act upon these definitions.